test/e2e_flags.go

```diff
@@ -58,22 +61,48 @@ type ServingEnvironmentFlags struct {
 	Https            bool // Indicates where the test service will be created with https
 }
 
-// initializeServingFlags registers flags used by e2e tests, calling flag.Parse() here would fail in
-// go1.13+, see https://github.com/knative/test-infra/issues/1329 for details
 func initializeServingFlags() *ServingEnvironmentFlags {
 	var f ServingEnvironmentFlags
 
+	// Only define flags here. Attempting flag.Set() will silently fail.
 	flag.BoolVar(&f.ResolvableDomain, "resolvabledomain", false,
 		"Set this flag to true if you have configured the `domainSuffix` on your Route controller to a domain that will resolve to your test cluster.")
 	flag.BoolVar(&f.Https, "https", false,
 		"Set this flag to true to run all tests with https.")
 
-	flag.Set("alsologtostderr", "true")
-	logging.InitializeLogger(test.Flags.LogVerbose)
-
-	if test.Flags.EmitMetrics {
-		logging.InitializeMetricExporter(E2EMetricExporter)
-	}
+	klog.InitFlags(nil)
 
 	return &f
 }
+
+var flagsSetup = false
+var flagsMutex = &sync.Mutex{}
+
+func SetupServingFlags() {
+	flagsMutex.Lock()
+	if !flagsSetup {
+		flag.Set("alsologtostderr", "true")
+
+		if test.Flags.LogVerbose {
+			// If klog verbosity is not set to a non-default value (via "-args -v=X"),
+			// set up verbosity for klog so round_trippers.go prints:
+			//   URL, request headers, response headers, and response body
+			// Definition in vendor/k8s.io/client-go/transport/round_trippers.go:DebugWrappers
+			if flag.CommandLine.Lookup("v").Value.String() == "0" {
+				flag.Set("v", "8")
+			}
+			fmt.Printf("Test Flags: {")
+			flag.CommandLine.VisitAll(func(f *flag.Flag) {
+				fmt.Printf("'%s': '%s', ", f.Name, f.Value.String())
+			})
+			fmt.Printf("}\n")
```

Comment:
Not using formatting
```suggestion
			fmt.Println("}")
```